brainBuildersRegistrationForm.php

```diff
@@ -9,550 +9,836 @@
+            <input type="hidden" name="id" value="<?php echo isset($_GET['id']) ? htmlspecialchars($_GET['id']) : htmlspecialchars($userID); ?>">
+
+            <!-- Looping through children to display the ones who have not completed the form -->
+            <?php 
+            $childrenNotRegistered = []; // Correctly initializing the array
+            foreach ($family_children as $child): 
+                require_once('database/dbBrainBuildersRegistrationForm.php');
+                if (!isBrainBuildersRegistrationFormComplete($child->getID())):
+                    // Add child to the array if they have not completed the form
+                    $childrenNotRegistered[] = $child;
+                endif;
+            endforeach;
+            ?>
+
+            <!--Child selection for children who have not yet completed a form-->
+            <select id="childDropdown" name="childId">
+                <option value="" disabled>Select</option>
+                <?php foreach ($childrenNotRegistered as $child): ?>
+                    <option value="<?php echo $child->getID(); ?>"
+                        <?php echo isset($_GET['childId']) && $_GET['childId'] == $child->getID() ? 'selected' : ''; ?>>
+                        <?php echo htmlspecialchars($child->getFirstName()) . " " . htmlspecialchars($child->getLastName()); ?>
+                    </option>
+                <?php endforeach; ?>
+            </select>
+
+            <input type="submit" value="Submit">
+        </form>
+
+        </div>
+
+        <br><hr><br>
+
+        <div id="formatted_form">
+
+        <?php
+        // If child ID is passed via GET, retrieve the child object
+        if (isset($_GET['childId'])) {
+            $selectedChild = retrieve_child_by_id($_GET['childId']);
+
+            // If child is found, creeate variables to populate the BB form with their details
+            if ($selectedChild) {
+                $child_id = $_GET['childId'];
+                $child_first_name = $selectedChild->getFirstName();
+                $child_last_name = $selectedChild->getLastName();
+                $child_gender = $selectedChild->getGender();
+                $child_school = $selectedChild->getSchool();
+                $child_grade = $selectedChild->getGrade();
+                $child_DOB = $selectedChild->getBirthDate();
+                $child_address = $selectedChild->getAddress();
+                $child_city = $selectedChild->getCity();
+                $child_state = $selectedChild->getState();
+                $child_zip = $selectedChild->getZip();
+                $child_medical = $selectedChild->getMedicalNotes();
+            }
+        } else {
+            $_POST['childId'] = null;
+            $_POST['child_first_name'] = null;
+            $_POST['child_last_name'] = null;
+        }
+        ?>
+
+        <form id="brainBuildersStudentRegistrationForm" action="" method="post">             
```

Comment:
Another UX thing, we should hide this part of the form until the user selects a child, since the user will lose all their progress once they choose a child

Reply:
ok, i just moved the php if statement that checks if &_GET['childId'] is set to be around the entire second part of the form